tutorials/terraform/data-migration-pgsql-mpg/data-restore-pgsql-mpg.tf

```diff
@@ -0,0 +1,162 @@
+  type        = set(string)
+  default = [
+    # Put the list of names of source database PostgreSQL extensions.
+    # Example:
+    # "pg_qualstats",
+    # "dblink"
+  ]
+}
+
+resource "yandex_vpc_network" "network" {
+  description = "Network for the Managed Service for PostgreSQL cluster and VM"
+  name        = "network"
+}
+
+resource "yandex_vpc_subnet" "subnet-a" {
+  description    = "Subnet in the ru-central1-a availability zone"
+  name           = "subnet-a"
+  zone           = "ru-central1-a"
+  network_id     = yandex_vpc_network.network.id
+  v4_cidr_blocks = ["10.1.0.0/24"]
+}
+
+resource "yandex_vpc_security_group" "security-group-mpg" {
+  description = "Security group for the Managed Service for PostgreSQL cluster"
+  network_id  = yandex_vpc_network.network.id
+}
+
+resource "yandex_vpc_security_group_rule" "rule-cluster" {
+  security_group_binding = yandex_vpc_security_group.security-group-mpg.id
+  direction              = "ingress"
+  description            = "Allow connections to the cluster from the Internet"
+  protocol               = "TCP"
+  port                   = 6432
+  v4_cidr_blocks         = ["0.0.0.0/0"]
+}
+
+# If you use VM for loading database dump and restoring data to the cluster, uncomment these lines.
+#resource "yandex_vpc_security_group" "security-group-vm" {
+#  description = "Security group for VM"
+#  network_id  = yandex_vpc_network.network.id
+#}
+#
+#resource "yandex_vpc_security_group_rule" "rule-vm-in" { 
+#  security_group_binding = yandex_vpc_security_group.security-group-vm.id
+#  direction              = "ingress"
+#  description            = "Allow SSH connections for VM from the Internet"
+#  protocol               = "TCP"
+#  port                   = 22
+#  v4_cidr_blocks         = ["0.0.0.0/0"]
+#}
+#
+#resource "yandex_vpc_security_group_rule" "rule-vm-out" {
+#  security_group_binding = yandex_vpc_security_group.security-group-vm.id
+#  direction              = "egress"
+#  description            = "Allow outgoing connections to any required resource"
+#  protocol               = "ANY"
+#  from_port              = 0
+#  to_port                = 65535
+#  v4_cidr_blocks         = ["0.0.0.0/0"]
+#}
+
+resource "yandex_mdb_postgresql_cluster" "mpg-cluster" {
+  description        = "Managed Service for PostgreSQL cluster"
+  name               = "mpg-cluster"
+  environment        = "PRODUCTION"
+  network_id         = yandex_vpc_network.network.id
+  security_group_ids = [yandex_vpc_security_group.security-group-mpg.id]
+
+  config {
+    version = local.target_pgsql_version
+    resources {
+      resource_preset_id = "s2.micro" # 2 vCPU, 8 GB RAM
+      disk_type_id       = "network-hdd"
+      disk_size          = 10 # GB
+    }
+  }
+
+  host {
+    zone             = "ru-central1-a"
+    subnet_id        = yandex_vpc_subnet.subnet-a.id
+    assign_public_ip = true # Required for connection from the Internet. For a method without intermediate VM.
```

Comment:
Required for connection from the Internet, for a method without an intermediate VM.